java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -2060,6 +2067,21 @@ public void changedUpdate(DocumentEvent e) {}
         return result;
     }
     
+    void updateJavaHintPreferences(JsonObject configuration) {
+        String path;
+        if (configuration != null) {
+            JsonElement pathPrimitive = configuration.get("preferences");
+            path = pathPrimitive.isJsonPrimitive() ? pathPrimitive.getAsString() : null;
```

Comment:
this code is not structured well. If path is not a primitive, does not exist or cannot be read, it still re-runs diagnostics for no reason. In addition - there's a series of nested if-s hidden in sequence of ternary expressions. 

Is there a reason to keep the path as String, convert it Path > File > URI > String, and then String > URI > FileObject - why it cannot be converted directly with `FileUtil.toFileObject()` and store that ?